apps/vaporgui/VLineEdit.cpp

```diff
@@ -0,0 +1,115 @@
+#include <iostream>
+#include <iomanip>
+#include <string>
+#include <sstream>
+
+#include <QMenu>
+
+#include "VLineEdit.h"
+#include "ErrorReporter.h"
+
+VLineEdit::VLineEdit( const std::string& value )
+: VContainer( this ),
+  _menu( nullptr ),
+  _value( value ),
+  _isDouble( false ),
+  _scientific( false ),
+  _menuEnabled( false ),
+  _decDigits( 4 )
+{
+    _lineEdit = new QLineEdit;
+    SetValue( _value );
+    layout()->addWidget(_lineEdit);
+
+    connect( _lineEdit, SIGNAL( editingFinished() ),
+        this, SLOT( emitLineEditChanged() ) );
+}
+
+void VLineEdit::UseMenu() {
+    _menu = new QMenu();
```

Comment:
If `_menu` was already pointed to an object, does this new operation cause a memory leak ? I recommend use std::unique_ptr<> for manual management of objects.

Reply:
Thanks for the catch.  QMenu is buggy if it uses Qt's memory management, so I'm going to elect to just keep this on the stack instead of heap.  This will let us avoid pointers and rule of 5.